onnxruntime/core/mlas/lib/aarch64/DepthwiseQConvKernelSize9Neon.S

```diff
@@ -0,0 +1,366 @@
+        eor     v22.16b, v22.16b, v5.16b
+        eor     v23.16b, v23.16b, v5.16b
+        
+        smlal   v24.8h, v14.8b, v20.8b
+        smlal2  v25.8h, v14.16b, v20.16b
+        smlal   v26.8h, v14.8b, v21.8b
+        smlal2  v27.8h, v14.16b, v21.16b
+        smlal   v28.8h, v15.8b, v22.8b
+        smlal2  v29.8h, v15.16b, v22.16b
+        smlal   v30.8h, v15.8b, v23.8b
+        smlal2  v31.8h, v15.16b, v23.16b
+
+        saddw   v16.4s, v16.4s, v24.4h
+        saddw2  v17.4s, v17.4s, v24.8h
+        saddw   v18.4s, v18.4s, v25.4h
+        saddw2  v19.4s, v19.4s, v25.8h
+
+        saddw   v6.4s, v6.4s, v26.4h
+        saddw2  v7.4s, v7.4s, v26.8h
+        saddw   v8.4s, v8.4s, v27.4h
+        saddw2  v9.4s, v9.4s, v27.8h
+
+        ldr     q10, [x1]                   // vk8
+        ldr     q20, [x28, x8]              // out0 vi8
+        ldr     q21, [x19, x8]              // out1 vi8
+
+        tbz     x6, #.LMLAS_CONV_SYM_FLAG_PER_CHANNEL_SCALE_BIT_INDEX, .LDonePerChannelScaleLoad_MlasConvSymDepthwiseKernelSize9
+        ld1     {v1.4s, v2.4s, v3.4s, v4.4s}, [x7], #64     	// scales 0-15 for outs
+
+.LDonePerChannelScaleLoad_MlasConvSymDepthwiseKernelSize9:
+        saddw   v16.4s, v16.4s, v28.4h
+        saddw2  v17.4s, v17.4s, v28.8h
+        eor     v20.16b, v20.16b, v5.16b
+        eor     v21.16b, v21.16b, v5.16b
+        saddw   v18.4s, v18.4s, v29.4h
+        saddw2  v19.4s, v19.4s, v29.8h
+
+        saddw   v6.4s, v6.4s, v30.4h
+        saddw2  v7.4s, v7.4s, v30.8h
+        saddw   v8.4s, v8.4s, v31.4h
+        saddw2  v9.4s, v9.4s, v31.8h
+
+        smull   v24.8h, v10.8b, v20.8b
+        smull2  v25.8h, v10.16b, v20.16b
+        smull   v26.8h, v10.8b, v21.8b
+        smull2  v27.8h, v10.16b, v21.16b
+
+        saddw   v16.4s, v16.4s, v24.4h
+        saddw2  v17.4s, v17.4s, v24.8h
+        saddw   v18.4s, v18.4s, v25.4h
+        saddw2  v19.4s, v19.4s, v25.8h
+
+        saddw   v6.4s, v6.4s, v26.4h
+        saddw2  v7.4s, v7.4s, v26.8h
+        saddw   v8.4s, v8.4s, v27.4h
+        saddw2  v9.4s, v9.4s, v27.8h
+
+        scvtf   v16.4s, v16.4s              // Requantize
+        scvtf   v17.4s, v17.4s
+        scvtf   v18.4s, v18.4s
+        scvtf   v19.4s, v19.4s
+        scvtf   v6.4s, v6.4s
+        scvtf   v7.4s, v7.4s
+        scvtf   v8.4s, v8.4s
+        scvtf   v9.4s, v9.4s
+
+        fmul    v16.4s, v16.4s, v1.4s
+        fmul    v17.4s, v17.4s, v2.4s
+        fmul    v18.4s, v18.4s, v3.4s
+        fmul    v19.4s, v19.4s, v4.4s
+        fmul    v6.4s, v6.4s, v1.4s
+        fmul    v7.4s, v7.4s, v2.4s
+        fmul    v8.4s, v8.4s, v3.4s
+        fmul    v9.4s, v9.4s, v4.4s
+
+        fcvtns  v16.4s, v16.4s
+        fcvtns  v17.4s, v17.4s
+        fcvtns  v18.4s, v18.4s
+        fcvtns  v19.4s, v19.4s
+        fcvtns  v6.4s, v6.4s
+        fcvtns  v7.4s, v7.4s
+        fcvtns  v8.4s, v8.4s
+        fcvtns  v9.4s, v9.4s
+
+        sqxtn   v16.4h, v16.4s              // +zp, narrow and combine
```

Comment:
there are lots of computation cycles here. Maybe we can shuffle register names here and leave v16-v20 alone, and then interleave the loading of vk0/1 vi0/1 registers, for the next iteration. in A75 CPUs those load instructions will be issued concurrently with other computations without costing anything.

Or, try changing to processing 8 channels, thus you use d formed load instructions instead of q formed ones. D form load instructions can be dual issued in A53 and A55, so their cycles can be completely hidden in A53 and A55 as well.

Reply:
Nice. Yet I also need to make code not too complex.